search/serializers.py

```diff
@@ -427,3 +525,72 @@ def serialize_course_for_bulk(course_obj):
         "_id": gen_course_id(course_obj.course_id),
         **ESCourseSerializer(course_obj).data,
     }
+
+
+def serialize_bulk_bootcamps(ids):
+    """
+    Serialize bootcamps for bulk indexing
+
+    Args:
+        ids(list of int): List of bootcamp id's
+    """
+    for bootcamp in Bootcamp.objects.filter(id__in=ids).prefetch_related("instructors"):
+        yield serialize_bootcamp_for_bulk(bootcamp)
+
+
+def serialize_bootcamp_for_bulk(bootcamp_obj):
+    """
+    Serialize a bootcamp for bulk API request
+
+    Args:
+        bootcamp_obj (Bootcamp): A bootcamp
+    """
+    return {
+        "_id": gen_bootcamp_id(bootcamp_obj.course_id),
+        **ESBootcampSerializer(bootcamp_obj).data,
+    }
+
+
+def serialize_bulk_programs(ids):
+    """
+    Serialize programs for bulk indexing
+
+    Args:
+        ids(list of int): List of program id's
+    """
+    for program in Program.objects.filter(id__in=ids):
+        yield serialize_program_for_bulk(program)
+
+
+def serialize_program_for_bulk(program_obj):
+    """
+    Serialize a program for bulk API request
+
+    Args:
+        program_obj (Program): A program
+    """
+    return {"_id": gen_program_id(program_obj), **ESProgramSerializer(program_obj).data}
+
+
+def serialize_bulk_user_lists(ids):
+    """
+    Serialize user_lists for bulk indexing
+
+    Args:
+        ids(list of int): List of user_list id's
+    """
+    for user_list in UserList.objects.filter(id__in=ids):
```

Comment:
I think this and all other `LearningResource` subclasses should `prefetch_related("topics")`

Reply:
Sure, I can make that change, but for it to have the intended effect (saving a database call), I'll have to change the downstream Serializer code containing `list(course.topics.values_list("name", flat=True))` to something like [topic.name for topic in course.topics.all()], as `values_list` call will put the burden on the database to collate the `name` field, thus making another database call despite the prefetching.

Reply:
Handled in https://github.com/mitodl/open-discussions/pull/2056/commits/70e5b4bcb79946a316496229befc67b58c1cc8f5